framework/wazuh/core/stats.py

```diff
@@ -2,23 +2,215 @@
+from wazuh.core import agent, common
+from wazuh.core.exception import WazuhError, WazuhException, WazuhInternalError, WazuhResourceNotFound
 from wazuh.core.wazuh_socket import WazuhSocket
 
+DAYS = "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"
+MONTHS = "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
+
+
+def hourly_():
+    """Compute hourly averages.
+
+    Returns
+    -------
+    array
+        Averages and iterations.
+    """
+    averages = []
+    interactions = 0
+    for i in range(25):
+        try:
+            hfile = open(f'{common.stats_path}/hourly-average/{str(i)}')
+            data = hfile.read()
+            if i == 24:
+                interactions = int(data)
+            else:
+                averages.append(int(data))
+            hfile.close()
+        except IOError:
+            if i < 24:
+                averages.append(0)
+
+    return [{'averages': averages, 'interactions': interactions}]
+
+
+def weekly_():
+    """Compute weekly averages.
+
+    Returns
+    -------
+    array
+        Hours and interactions for each week day.
+    """
+    weekly_results = []
+    for i in range(7):
+        hours = []
+        interactions = 0
+        for j in range(25):
+            try:
+                wfile = open(f'{common.stats_path}/weekly-average/{str(i)}/{str(j)}')
+                data = wfile.read()
+                if j == 24:
+                    interactions = int(data)
+                else:
+                    hours.append(int(data))
+                wfile.close()
+            except IOError:
+                if i < 24:
+                    hours.append(0)
+        weekly_results.append({DAYS[i]: {'hours': hours, 'interactions': interactions}})
+
+    return weekly_results
+
+
+def totals_(date):
+    """Statistical information for the current or specified date.
+
+    Parameters
+    ----------
+    date: date
+        Date object with the date value of the stats.
+
+    Returns
+    -------
+    tuple
+        Failed: boolean that represents if data parsing failed. Its value is always `False`.
+        Affected: array of dictionaries. Each dictionary represents an hour.
+
+    Raises
+    ------
+    WazuhError
+        Raised on `IOError`.
+    """
+    try:
+        stat_filename = os.path.join(
+            common.stats_path, "totals", str(date.year), MONTHS[date.month - 1],
+            f"ossec-totals-{date.strftime('%d')}.log")
+        stats = open(stat_filename, 'r')
+    except IOError:
+        raise WazuhError(1308, extra_message=stat_filename)
+    alerts = []
+    affected = []
+    for line in stats:
+        data = line.split('-')
+        if len(data) == 4:
+            sigid = int(data[1])
+            level = int(data[2])
+            times = int(data[3])
+            alert = {'sigid': sigid, 'level': level, 'times': times}
+            alerts.append(alert)
+        else:
+            data = line.split('--')
```

Comment:
I don't fully understand the purpose of this piece of code and why do we need to return a tuple with a boolean when it fails, can't we use standard exception raise/catch?

Reply:
This code block stands for parsing 'totals' stats file. The bool in returned tuple is used in 'stats.py` (sdk) and yes, it could be changed to exception management. I thought this way (with boolean) would be simpler but I can arrange those changes if needed.